x/dex/keeper/msgserver/msg_server_register_contract.go

```diff
@@ -242,6 +228,9 @@ func (k msgServer) SetNewContract(ctx sdk.Context, msg *types.MsgRegisterContrac
 		dependency.ImmediateYoungerSibling = ""
 		found := false
 		for _, contractInfo := range allContractInfo {
+			if contractInfo.ContractAddr == newContract.ContractAddr {
```

Comment:
this was not order agnostic previously @codchen and it updated the newContract to its dependency in a wrong order